### PR TITLE
[autoscaler] Improve Logtimer log messages

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?

Logtimer currently prints 'conflicting' messages when a task fails.  It prints the default message, regardless of what the outcome is. The new behavior prints a different log if the timed method/code fails. 

Current Behavior (on failure): `Ray start commands completed`
New Behavior (on failure):  `Ray start commands FAILED`

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/autoscaler/log_timer.py [hunk 1]
```diff
@@ -5,13 +5,18 @@
...
     def __enter__(self):
         self._start_time = datetime.datetime.utcnow()
 
-    def __exit__(self, *_):
+    def __exit__(self, *error_vals):
         td = datetime.datetime.utcnow() - self._start_time
-        logger.info(self._message +
-                    " [LogTimer={:.0f}ms]".format(td.total_seconds() * 1000))
+        if any(error_vals):
+            self._status_string = "FAILED"
```
Comment (richardliaw) [inline python/ray/autoscaler/log_timer.py:18]: `failed` instead of `FAILED`?

File: python/ray/autoscaler/updater.py [hunk 2]
```diff
@@ -423,16 +423,16 @@ def do_update(self):
...
                 self.node_id, {TAG_RAY_NODE_STATUS: STATUS_SETTING_UP})
-            with LogTimer(self.log_prefix +
-                          "Initialization commands completed"):
+            with LogTimer(self.log_prefix + "Initialization commands",
+                          "completed"):
                 for cmd in self.initialization_commands:
                     self.cmd_runner.run(cmd)
 
-            with LogTimer(self.log_prefix + "Setup commands completed"):
+            with LogTimer(self.log_prefix + "Setup commands", "completed"):
```
Comment (richardliaw) [inline python/ray/autoscaler/updater.py:431]: any reason to have an explicit success str?

File: python/ray/autoscaler/log_timer.py (hunk 1 above)
Comment (ijrsvt) [reply python/ray/autoscaler/log_timer.py:18]: Sounds good!

File: python/ray/autoscaler/updater.py (hunk 2 above)
Comment (ijrsvt) [reply python/ray/autoscaler/updater.py:431]: Not really--are you suggesting that all commands have "success" upon completion and "failed" upon failure?

Comment (richardliaw) [reply python/ray/autoscaler/updater.py:431]: yeah, maybe flag it off with `LogTimer(msg, show_status=True)`

Comment (ijrsvt) [reply python/ray/autoscaler/updater.py:431]: Good idea!

**Comment by richardliaw** [comment]
Are there other uses of LogTimer?

File: python/ray/autoscaler/log_timer.py [hunk 3]
```diff
@@ -5,13 +5,19 @@
 
 
 class LogTimer:
-    def __init__(self, message):
+    def __init__(self, message, show_status=True):
```
Comment (richardliaw) [inline python/ray/autoscaler/log_timer.py:8]: if there's other usage of LogTimer, you'll probably want to default this to False

Comment (ijrsvt) [reply python/ray/autoscaler/log_timer.py:8]: Sure!

**Comment by ijrsvt** [comment]
@richardliaw 
I could change all the `LogTimer` cases to go from: `___ed <noun>`  to `____ing <failed/succeeded`. 

This would avoid cases where both :
`Got remote shell  [LogTimer=1004ms]` and `Unable to connect to node` are printed. 

This would change it to 
`Getting remote shell  failed [LogTimer=1004ms]` and `Unable to connect to node` are printed.

File: python/ray/autoscaler/updater.py [hunk 4]
```diff
@@ -423,16 +423,15 @@ def do_update(self):
             # Run init commands
             self.provider.set_node_tags(
                 self.node_id, {TAG_RAY_NODE_STATUS: STATUS_SETTING_UP})
-            with LogTimer(self.log_prefix +
-                          "Initialization commands completed"):
+            with LogTimer(self.log_prefix + "Initialization commands", True):
```
Comment (simon-mo) [inline python/ray/autoscaler/updater.py:426]: can you add the keyword label so by just looking at this piece of code we understand what's going on? `show_status=True`

Comment (ijrsvt) [reply python/ray/autoscaler/updater.py:426]: Good idea!

**Review by simon-mo — APPROVED** [review]

**Pull request merged**